test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -639,9 +645,8 @@ func unrollWorkloadTemplate(b *testing.B, wt []op, w *workload) []op {
 	return unrolled
 }
 
-func runWorkload(b *testing.B, tc *testCase, w *workload) []DataItem {
-	// 30 minutes should be plenty enough even for the 5000-node tests.
-	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Minute)
+func runWorkload(parentCtx context.Context, b *testing.B, tc *testCase, w *workload) []DataItem {
+	ctx, cancel := context.WithCancel(parentCtx)
```

Comment:
Why do we need a sub context? Can't we just use `parentCtx` in all cases? If so, it should just be renamed to `ctx`.

Reply:
Also discussed with @sanposhiho about this https://github.com/kubernetes/kubernetes/pull/114843#discussion_r1070815242